### PR TITLE
Added CollectorTaskWithGenericItem extending CollectorTask.

### PR DESCRIPTION
**Pull request by tabladrum**
Collectors that need to process generic items created by jenkins plugin, can extend this new Abstract class that has the the method to process generic item for a given collector.

### CODE REVIEW

File: src/main/java/com/capitalone/dashboard/collector/CollectorTaskWithGenericItem.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        this.relatedCollectorItemRepository = relatedCollectorItemRepository;
+    }
+
+
+    public Map<ObjectId, Set<ObjectId>> processGenericItems(List<String> toolServers) {
+        List<GenericCollectorItem> genericCollectorItems = genericCollectorItemRepository.findAllByToolNameAndProcessTimeEquals(getCollector().getName(), 0L);
+        Map<ObjectId, Set<ObjectId>> collectorItemBuildIds = new HashMap<>();
+        genericCollectorItems.forEach(gci -> {
+            toolServers.stream()
+                    .map(server -> Lists.newArrayList(collectorItemRepository.findAllByOptionMapAndCollectorIdsIn(getGenericCollectorItemOptions(server, gci), Lists.newArrayList(getCollector().getId()))))
```
Comment (Sbrenthughes) [inline src/main/java/com/capitalone/dashboard/collector/CollectorTaskWithGenericItem.java]: Can  `collectorItemRepository.findAllByOptionMapAndCollectorIdsIn` ever return null?

Comment (tabladrum) [reply src/main/java/com/capitalone/dashboard/collector/CollectorTaskWithGenericItem.java]: +1

Comment (tabladrum) [reply src/main/java/com/capitalone/dashboard/collector/CollectorTaskWithGenericItem.java]: it does not, but it could change in spring querydsl !

Comment (tabladrum) [reply src/main/java/com/capitalone/dashboard/collector/CollectorTaskWithGenericItem.java]: See this test case for empty repository:
```
    @Test
    public void processGenericItemsEmptyGenericItemRepo() throws IOException {
        reset();

        Collector collector = new Collector();
        collector.setId(new ObjectId("5ba136290be2d32568777fa8"));
        collector.setName("Sonar");
        Map<ObjectId, Set<ObjectId>> objectIdSetMap = testCollectorTaskWithGenericItem.processGenericItems(Lists.newArrayList("http://localhost:9000"));
        List<RelatedCollectorItem> relatedCollectorItems = Lists.newArrayList(relatedCollectorItemRepository.findAll());
        assertEquals(0, objectIdSetMap.size());
        assertEquals(0, relatedCollectorItems.size());
    }
```

**Pull request merged**